### PR TITLE
py-bluecellulab: add package

### PR DESCRIPTION
**Pull request by arnaudon**

### CODE REVIEW

File: bluebrain/repo-bluebrain/packages/py-bluecellulab/package.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    version("develop", branch="main")
+    version("1.5.2", commit="22e45b921ad9cd282d53b57ac90fcec69731cb718ed8098c8eb52039bf7a49e5")
+
+    depends_on("py-setuptools", type=("build", "run"))
+    depends_on("neuron+python", type=("build", "run"))
+
+    # dependencies from setup.py
+    depends_on("py-numpy@1.8:", type=("build", "run"))
+    depends_on("py-matplotlib@3.0.0:", type=("build", "run"))
+    depends_on("py-cachetools", type=("build", "run"))
```
Comment (anilbey) [inline bluebrain/repo-bluebrain/packages/py-bluecellulab/package.py]: This dependency has been removed. You can remove it here also.

**Review by anilbey — CHANGES_REQUESTED** [review]
There are these 3 new dependencies.

```
        "bluepysnap>=1.0.5,<2.0.0",
        "pydantic>=1.10.2,<2.0.0",
        "typing-extensions>=4.8.0"
```

and the py-cachetools as well as bluepy-configfile can be removed safely

File: bluebrain/repo-bluebrain/packages/py-bluecellulab/package.py (hunk 1 above)
Comment (arnaudon) [reply bluebrain/repo-bluebrain/packages/py-bluecellulab/package.py]: done!

**Review by anilbey — DISMISSED** [review]
Looks very good to me, thanks!

File: bluebrain/deployment/environments/applications_science.yaml [hunk 2]
```diff
@@ -55,6 +56,7 @@ spack:
     - py-bba-data-push
     - py-bba-webexporter
     - py-bglibpy
```
Comment (matz-e) [inline bluebrain/deployment/environments/applications_science.yaml]:
```suggestion
```

File: bluebrain/deployment/environments/applications_science.yaml [hunk 3]
```diff
@@ -12,6 +12,7 @@ spack:
           - py-bba-data-push
           - py-bba-webexporter
           - py-bglibpy
```
Comment (matz-e) [inline bluebrain/deployment/environments/applications_science.yaml]:
```suggestion
```

**Review by matz-e — COMMENTED** [review]
While we're at it, why not stop deploying `bglibpy`? I'll probably have to patch up the module tests, though.

**Comment by matz-e** [comment]
@arnaudon we have this test for `bglibpy` that I tried to "massage" into a hybrid that would work with `bluecellulab`:

https://bbpgitlab.epfl.ch/hpc/module-testing/-/blob/main/scripts/bglibpy_test.py?ref_type=heads

Would you be able to help me fix that error from the module testing?
```
Mod lib:  No mechanisms are loaded.
Traceback (most recent call last):
  File "/gpfs/bbp.cscs.ch/ssd/gitlab_map_jobs/bbpcihpcproj12/J955664/hpc/module-testing/scripts/bglibpy_test.py", line 8, in <module>
    soma.insert('Ca_HVA2')
ValueError: argument not a density mechanism name.
```

Full test script:
```python
try:
    import bglibpy
    neuron = bglibpy.neuron
except:
    import bluecellulab
    neuron = bluecellulab.neuron
soma = neuron.h.Section()
soma.insert('Ca_HVA2')
neuron.h.load_file('stdrun.hoc')
neuron.h.run()
print(soma.v)
vecstim = neuron.h.VecStim()
```

**Comment by anilbey** [comment]
Hi @matz-e how are you loading the mechanisms before loading the script? Are you using the `BGLIBPY_MOD_LIBRARY_PATH` environment variable or is NEURON loading the mechanisms in your cwd at the time you write the import statement?

**Comment by anilbey** [comment]
I could run your script with the compiled mod files provided.

```
❯ python spack-test.py
        1 
Imported NEURON from: /VENV_PATH/bluecellulab311/lib/python3.11/site-packages/neuron/__init__.py
Mod lib:  /MOD_PATH/BlueCelluLab/x86_64
-64.99999985986676

```

The way of loading mechanisms is changed in bluecellulab. It is now relying on NEURON's API for doing so. 

It is described in here: https://bluecellulab.readthedocs.io/en/latest/compiling-mechanisms.html

**Review by matz-e — APPROVED** [review]

**Comment by arnaudon** [comment]
Thank you for you help on that one!

**Pull request merged**